### PR TITLE
Command Handler with subfolders

### PR DESCRIPTION
**Pull request by BannukDE**
New to JS and GitHub, still thought you might want to look @it.

Hope everything attaches correctly ;)

OneSentenceDescription:
Recursive path search with 2-dimensional-array as return to get all #commands out of subfolders.

### CODE REVIEW

**Review by BannukDE — COMMENTED** [review]
Added a third commit for resolving errors and cleaningCode

**Comment by YorkAARGH** [comment]
You removed a required definition of `props` on line 48 which will crash the bot, I'm currently consulting with @eslachance about this PR.

**Comment by BannukDE** [comment]
Found a last "props" in line 65 needs to be attached to "client.props", also change "var CmdFiles" to "const CmdFiles" in line 47.

In addition all CmdFiles NEED to be in Subfolder.

**Comment by eslachance** [comment]
Please move the commands back to the root `./commands/` and make them work from there, in order to ensure that *existing* bots wanting to update, do not crash. Also, these are not "Example" commands. Those are literal, working, useful commands that people can use. 

Do not confuse this repository with your own work space. If you want those commands to be moved in your bot, do it in your own fork, not in this repository.

File: index.js [hunk 1]
```diff
@@ -39,20 +39,34 @@ client.settings = new PersistentCollection({name: "settings"});
...
+  const cmdFiles = await client.pathwalker("./commands/");
+  client.log("log", `Loading a total of ${cmdFiles[1].length} commands out of ${cmdFiles[0].length} folders.`);
+  // for each CMD
+  cmdFiles[1].forEach(f => {
     try {
-      const props = require(`./commands/${f}`);
+      // try each Path
+      cmdFiles[0].forEach(d =>{
+        try {
+          client.props = require(`./commands/${d}/${f}`);
```
Comment (eslachance) [inline index.js]: No. `client.props` is global, that's absolutely horrifyingly stupid. `const props` must be a LOCALLY SCOPED VAR. putting it on client makes no goddamn sense.

File: index.js [hunk 2]
```diff
@@ -39,20 +39,34 @@ client.settings = new PersistentCollection({name: "settings"});
...
+  */
+  const cmdFiles = await client.pathwalker("./commands/");
+  client.log("log", `Loading a total of ${cmdFiles[1].length} commands out of ${cmdFiles[0].length} folders.`);
+  // for each CMD
+  cmdFiles[1].forEach(f => {
     try {
-      const props = require(`./commands/${f}`);
+      // try each Path
+      cmdFiles[0].forEach(d =>{
+        try {
```
Comment (eslachance) [inline index.js]: no no no no no no

Why are you adding a try/catch inside a try/catch. WHY.

File: index.js [hunk 3]
```diff
@@ -39,20 +39,34 @@ client.settings = new PersistentCollection({name: "settings"});
...
-  // here and everywhere else.
-  const cmdFiles = await readdir("./commands/");
-  client.log("log", `Loading a total of ${cmdFiles.length} commands.`);
-  cmdFiles.forEach(f => {
+  /*
+  Here we load **commands** into memory, as a collection,
+  so they're accessible here and everywhere else.
+  Useing the pathwalker function out of functions.js.
+  */
+  const cmdFiles = await client.pathwalker("./commands/");
```
Comment (eslachance) [inline index.js]: Why add a function to the client if you're only going to use it once? You should only make functions that you're going to use multiple times.

**Review by eslachance — CHANGES_REQUESTED** [review]
That, and reverting the "examples" folder crap.

**Comment by BannukDE** [comment]
I can understand that it must be frustrating to read the code of someone bad. But either you could find a nicer way telling me that my code creates "Brain anorisms" or I will take my first sentence: "...thought you might want to look..." literally and will neither work nor correspond anymore, with you.

If you can find, how do you like the changes?
I think I changed every requested object.

**Comment by eslachance** [comment]
I know you're trying to be helpful, and I appreciate the effort you want to make, however, you're simply reducing the quality of the code with your PR. You went from attaching a temporary variable to the client, to using readDirSync, which is **blocking**, instead of using promises. 

I'm going to have to close this PR, as it's currently going nowhere and will not be helpful to GuideBot.

**Pull request closed without merging**